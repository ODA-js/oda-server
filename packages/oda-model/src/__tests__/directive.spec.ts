import 'jest';
import { Directive } from '..';

describe('directive', () => {
  it('loads', () => {
    const res = new Directive({
      name: 'ReadOnly',
      on: ['FIELD'],
      args: [
        {
          name: 'field',
          type: 'string',
          required: false,
          defaultValue: 'default',
        },
      ],
    });
    expect(res).toMatchSnapshot('raw');
    expect(res.toObject()).toMatchSnapshot('toObject');
  });
});
