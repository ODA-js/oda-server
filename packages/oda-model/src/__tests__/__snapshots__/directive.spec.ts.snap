// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`directive loads: raw 1`] = `
Directive {
  Symbol(internal): Object {
    "args": Map {
      "field" => Object {
        "defaultValue": "default",
        "name": "field",
        "required": false,
        "type": "string",
      },
    },
    "name": "readOnly",
    "on": Array [
      "FIELD",
    ],
  },
  Symbol(metadata): Object {},
}
`;

exports[`directive loads: toObject 1`] = `
Object {
  "args": Array [
    Object {
      "defaultValue": "default",
      "name": "field",
      "required": false,
      "type": "string",
    },
  ],
  "description": undefined,
  "metadata": Object {},
  "name": "readOnly",
  "on": Array [
    "FIELD",
  ],
  "title": undefined,
}
`;
